components/consumable.py

```diff
@@ -91,17 +91,19 @@ def activate(self, action: actions.ItemAction) -> None:
         target_xy = action.target_xy
 
         if not self.engine.game_map.visible[target_xy]:
+            self.engine.clear_aoe()
             raise Impossible("You cannot target an area that you cannot see.")
 
         targets_hit = False
         for actor in self.engine.game_map.actors:
-            if actor.distance(*target_xy) <= self.radius:
+            if self.engine.game_map.aoe[actor.x, actor.y] == True:
```

Comment:
```suggestion
            if self.engine.game_map.aoe[actor.x, actor.y]:
```
Note that with the previous handling of `aoe` this check causes targets that are in-range but out-of-view to not be hit.  I'd consider this a bug.

Reply:
It might be better to generate a local `aoe` array in this function instead of using `self.engine.game_map.aoe`

Reply:
This was actually a decision on my part to implement it that way since you could not target area out of your view.

To fix it, just merge the aoe generated from the fov calculation and merge it with explored instead of visible.  You will see the aoe preview on explored areas, but you still will not be able to target any tile out of your view though.  This way It does handle the case of  being able to hit an entity that moved just out of your view but still close enough where the center of the blast is in view.